common/laws/00_internal_security.txt

```diff
@@ -105,39 +158,116 @@ law_national_guard = {
 law_secret_police = {
 	group = lawgroup_internal_security
 	icon = "gfx/interface/icons/law_icons/secret_police.dds"
-	
-	progressiveness = 0
-	
+	progressiveness = 25
 	unlocking_technologies = {
 		central_archives
 	}
+	unlocking_laws = {
+		law_autocracy
+		law_patriot_schools
+		law_militarized_police
+		law_dedicated_police
+	}
 
+	disallowing_laws = {
+		law_right_of_assembly
+		law_protected_speech
+	}
 	modifier = {
-		country_institution_home_affairs_max_investment_add = 2
+		#country_institution_home_affairs_max_investment_add = 2
 	}
-	
 	institution = institution_home_affairs
 	institution_modifier = {
 		country_suppression_ig_attraction_mult = -0.05
-		political_movement_radicalism_mult = -0.1
+		country_authority_add = 30
+		country_tech_spread_mult = 0.03
+		country_radicals_from_legitimacy_mult = -0.03
+		interest_group_ig_intelligentsia_pol_str_mult = 0.01
+		interest_group_ig_petty_bourgeoisie_pol_str_mult = 0.04
+		country_secession_progress_mult = 0.1
+
 	}
-	
 	possible_political_movements = {
 		law_guaranteed_liberties
+		law_empowered_gendarmerie
+		law_provincial_courts
+		law_inquisition
 	}
-
 	pop_support = {
 		value = 0
 	}
+	ai_enact_weight_modifier = {
+		#Petitions
+		value = 0
+		if = {
+			limit = {
+				has_journal_entry = je_government_petition
+				has_variable = desired_law_var
+				scope:law = var:desired_law_var
+			}
+			add = 750
+		}
+	}
+}
 
-	ai_enact_weight_modifier = { #Petitions
+law_empowered_gendarmerie = {
+	group = lawgroup_internal_security
+	icon = "gfx/interface/icons/law_icons/secret_police.dds"
+	progressiveness = 25
+	unlocking_technologies = {
+		nationalism
+	}
+	unlocking_laws = {
+		law_ethnostate
+		law_national_supremacy
+		law_patriot_schools
+		law_militarized_police
+		law_dedicated_police
+	}
+	disallowing_laws = {
+		law_protected_speech
+	}
+	modifier = {
+		#country_institution_home_affairs_max_investment_add = 2
+	}
+	institution = institution_home_affairs
+	institution_modifier = {
+		country_authority_add = 25
+		political_movement_radicalism_mult = -0.05
+		state_radicals_from_sol_change_mult = -0.025
+		state_loyalists_from_sol_change_mult = 0.01
+		state_turmoil_effects_mult = -0.04
```

Comment:
This is really powerful, I don't know if balanced but this internal security is very good.

Reply:
What makes it powerful in your opinion?

Reply:
Made conditions more restrictive to balance